dali/operators/generic/one_hot.h

```diff
@@ -102,12 +102,10 @@ class OneHot : public Operator<Backend> {
     for (int i = 0; i < num_samples; i++) {
       output_desc[0].shape.set_tensor_shape(i, determine_shape(shape[i], output_sample_dim));
     }
-    TYPE_SWITCH(output_type_, type2id, DType, ONE_HOT_TYPES, ({
-                  TypeInfo type;
-                  type.SetType<DType>(output_type_);
-                  output_desc[0].type = type;
-                }),
+    TYPE_SWITCH(output_type_, type2id, DType, ONE_HOT_TYPES, (
```

Comment:
So now this switch is only for supported type validation, right?

Reply:
Yes. I guess it could be done more elegantly, but I don't know if this is strictly necessary in this PR.